pgcli/packages/sqlcompletion.py

```diff
@@ -27,6 +28,15 @@ def suggest_type(full_text, text_before_cursor):
 
     identifier = None
 
+    def strip_named_query(txt):
+        pattern = re.compile(r'\\(n|nd|ns\s+[a-z]+)\s+')
```

Comment:
You might want to anchor it to the beginning of the query so it doesn't match `\n or \nd` in the middle of the query.